components/openindiana/nvidia-304/Makefile

```diff
@@ -0,0 +1,202 @@
+PKG_OPTIONS +=	-D NVIDIA_MEDIATOR="mediator=nvidia-drv mediator-implementation=nvidia-drv-proprietary mediator-version=$(NVIDIA_DRIVER_MAJOR_VERSION)"
+
+COMPONENT_POST_INSTALL_ACTION = ( \
+	$(CHMOD) u+w $(PROTO_DIR)/usr/X11/lib/modules/NVIDIA-$(NVIDIA_DRIVER_MAJOR_VERSION)/libnvidia-wfb.so.1 ; \
+	$(CHMOD) u+w $(PROTO_DIR)/usr/X11/lib/modules/NVIDIA-$(NVIDIA_DRIVER_MAJOR_VERSION)/$(MACH64)/libnvidia-wfb.so.1 ; \
+	/usr/bin/elfedit -e "dyn:rpath $(GCC_ROOT)/lib:/lib:/usr/lib" \
+		$(PROTO_DIR)/usr/X11/lib/modules/NVIDIA-$(NVIDIA_DRIVER_MAJOR_VERSION)/libnvidia-wfb.so.1 ; \
+	/usr/bin/elfedit -e "dyn:rpath $(GCC_ROOT)/lib/$(MACH64):/lib/$(MACH64):/usr/lib/$(MACH64)"\
+		$(PROTO_DIR)/usr/X11/lib/modules/NVIDIA-$(NVIDIA_DRIVER_MAJOR_VERSION)/$(MACH64)/libnvidia-wfb.so.1 ; )
+
+$(SOURCE_DIR)/.unpacked: download Makefile $(PATCHES)
+	$(RM) -r $(SOURCE_DIR)
+	$(SHELL) $(USERLAND_ARCHIVES)$(COMPONENT_ARCHIVE) -x
+	$(TOUCH) $@
+
+$(BUILD_32):	$(SOURCE_DIR)/.prep
+	$(RM) -r $(@D) ; $(MKDIR) $(@D)
+	$(TOUCH) $@
+
+check_protodir_32:	$(BUILD_32)
+	[ -n "$(PROTO_DIR)" ] && [ x"$(PROTO_DIR)" != x/ ] || exit
+
+$(BUILD_DIR_32)/.installed_drv:	check_protodir_32
+	$(RM) -r "$(PROTO_DIR)/kernel"
+	[ -d "$(PROTO_DIR)/kernel/drv/$(MACH64)" ] || $(MKDIR) "$(PROTO_DIR)/kernel/drv/$(MACH64)"
+	for i in \
+	    kernel/drv/$(MACH64)/nvidia \
+	; do \
+	    cp "$(SOURCE_DIR)/NVDAgraphicsr/reloc/$$i" \
+	        "$(PROTO_DIR)/$$i-$(NVIDIA_DRIVER_MAJOR_VERSION)" \
+	    && ln -s "`basename $$i`-$(NVIDIA_DRIVER_MAJOR_VERSION)" "$(PROTO_DIR)/$$i" \
+	    || exit ; \
+	done
+	cp "$(SOURCE_DIR)/NVDAgraphicsr/reloc/kernel/drv/nvidia.conf" \
+	    "$(PROTO_DIR)/kernel/drv/nvidia-$(NVIDIA_DRIVER_MAJOR_VERSION).conf" \
+	    && ln -s "nvidia-$(NVIDIA_DRIVER_MAJOR_VERSION).conf" \
+	        "$(PROTO_DIR)/kernel/drv/nvidia.conf"
+	$(TOUCH) $@
+
+install_drv_32: $(BUILD_DIR_32)/.installed_drv
+
+$(BUILD_DIR_32)/.installed_reloc:	check_protodir_32
+	$(RM) -r "$(PROTO_DIR)/usr"
+	cp -a "$(SOURCE_DIR)/NVDAgraphics/reloc" "$(PROTO_DIR)/usr"
+	$(TOUCH) $@
+
+install_reloc_32: $(BUILD_DIR_32)/.installed_reloc
+
+install_mv_bigdirs_32:	install_reloc_32
+	for i in \
+	    share/doc/NVIDIA \
+	    share/icons/NVIDIA \
+	    X11/lib/NVIDIA \
+	    X11/lib/modules/extensions/NVIDIA \
+	    X11/lib/modules/NVIDIA \
+	; do \
+	    mv -f "$(PROTO_DIR)/usr/$$i" \
+	        "$(PROTO_DIR)/usr/$$i-$(NVIDIA_DRIVER_MAJOR_VERSION)" \
+	    && ln -s "`basename $$i`-$(NVIDIA_DRIVER_MAJOR_VERSION)" "$(PROTO_DIR)/usr/$$i" \
+	    || exit ; \
+	done
+	$(MKDIR) "$(PROTO_DIR)/usr/share/nvidia-$(NVIDIA_DRIVER_MAJOR_VERSION)" \
+	    && ln -s "nvidia-$(NVIDIA_DRIVER_MAJOR_VERSION)" "$(PROTO_DIR)/usr/share/nvidia"
+
+install_mv_desktop_32:	install_reloc_32
+	mv -f "$(PROTO_DIR)/usr/share/applications/nvidia-settings.desktop" \
+	    "$(PROTO_DIR)/usr/share/applications/nvidia-settings-$(NVIDIA_DRIVER_MAJOR_VERSION).desktop" \
+	    && ln -s "nvidia-settings-$(NVIDIA_DRIVER_MAJOR_VERSION).desktop" \
```

Comment:
I suppose this will not work as expected, we'll have several desktop files instead of one.